### PR TITLE
General updates for health-kick

### PR DESCRIPTION
**Pull request by mattops**
- Update packages, helm charts, circleci orb
- Bring code up-to-date - and remove unused configuration, and tighten up rules.
- Restrict access to pingdom and internal IP addreses only.

### CODE REVIEW

**Review by psoleckimoj — APPROVED** [review]
LGTM!

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 version: 2.1
 
 orbs:
-  hmpps: ministryofjustice/hmpps@5.1
+  hmpps: ministryofjustice/hmpps@7.4
```
Comment (petergphillips) [inline .circleci/config.yml:4]: we normally float on minor, so `@7`

File: Dockerfile [hunk 2]
```diff
@@ -1,7 +1,10 @@
-FROM node:16.16-bullseye as builder
+FROM node:18.17-bullseye-slim as builder
```
Comment (petergphillips) [inline Dockerfile:1]: odd choice, latest is `18.18`

File: .circleci/config.yml (hunk 1 above)
Comment (petergphillips) [reply .circleci/config.yml:4]: also worth ensuring that this file uses 18.18-browsers too

File: Dockerfile [hunk 3]
```diff
@@ -1,7 +1,10 @@
...
 
-ARG BUILD_NUMBER
-ARG GIT_REF
+ARG BUILD_NUMBER=1_0_0
```
Comment (petergphillips) [inline Dockerfile:3]: dubious about this change - why default the value?  Should always be passed in and considered to be an error if it is not.

File: Dockerfile [hunk 4]
```diff
@@ -1,7 +1,10 @@
...
+ARG GIT_REF=not-available
+
+ENV BUILD_NUMBER ${BUILD_NUMBER:-1_0_0}
```
Comment (petergphillips) [inline Dockerfile:6]: would prefer this not to be defaulted.  Also you're using deprecated syntax, new version should be
```
ENV BUILD_NUMBER=${BUILD_NUMBER}
```

File: helm_deploy/health-kick/values.yaml [hunk 5]
```diff
@@ -25,5 +23,11 @@ generic-service:
...
+    groups:
+      - internal
```
Comment (petergphillips) [inline helm_deploy/health-kick/values.yaml:30]: hadn't realised that you could combine `groups` and an allowlist - nice

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**